.eslintrc.json

```diff
@@ -0,0 +1,15 @@
+{
+  "extends": [
+    "xo-typescript",
+    "plugin:import/recommended",
+    "plugin:import/typescript",
+    "plugin:unicorn/recommended",
+    "plugin:prettier/recommended"
+  ],
+  "root": true,
+  "rules": {
+    // These are incompatible with TypeScript "module: commonjs"
+    "@typescript-eslint/no-floating-promises": "off",
+    "unicorn/prefer-top-level-await": "off"
+  }
+}
```

Comment:
Does it make sense to apply `eslint --fix` in this commit? Usually we do that when configuring a linter.

Reply:
After `eslint --fix` we'll still have 16 non-auto-fixable (and frankly non-trivial even with human intervention) linter errors. I'd wager probably not worth rolling back to try to fix that since the code being fixed gets fully replaced in the next commits anyway, but that's not a hill I'd die on or anything.

```
(woods) github-actions-zulip  » (0cfc8ee...) » git checkout 0cfc8eec86b7719aed832f13bd3528c92615d5ef; npm ci; npx eslint --fix ./src/send-message.js 
HEAD is now at 0cfc8ee deps: Install and configure ESLint based on XO defaults.

added 278 packages, and audited 279 packages in 3s

# snip...

/home/j/src/zulip/github-actions-zulip/src/send-message.js
  29:13  error  Unsafe assignment of an `any` value                @typescript-eslint/no-unsafe-assignment
  29:31  error  Unsafe call of an `any` typed value                @typescript-eslint/no-unsafe-call
  29:50  error  Unsafe return of an `any` typed value              @typescript-eslint/no-unsafe-return
  29:50  error  Unsafe call of an `any` typed value                @typescript-eslint/no-unsafe-call
  31:9   error  Unsafe assignment of an `any` value                @typescript-eslint/no-unsafe-assignment
  31:14  error  Unsafe call of an `any` typed value                @typescript-eslint/no-unsafe-call
  49:11  error  Unsafe assignment of an `any` value                @typescript-eslint/no-unsafe-assignment
  49:26  error  Unsafe call of an `any` typed value                @typescript-eslint/no-unsafe-call
  57:11  error  Unsafe assignment of an `any` value                @typescript-eslint/no-unsafe-assignment
  57:28  error  Unsafe call of an `any` typed value                @typescript-eslint/no-unsafe-call
  60:50  error  Invalid type "any" of template literal expression  @typescript-eslint/restrict-template-expressions
  62:13  error  Unsafe assignment of an `any` value                @typescript-eslint/no-unsafe-assignment
  63:14  error  Invalid type "any" of template literal expression  @typescript-eslint/restrict-template-expressions
  63:32  error  Invalid type "any" of template literal expression  @typescript-eslint/restrict-template-expressions
  67:5   error  Implicit any in catch clause                       @typescript-eslint/no-implicit-any-catch
  75:5   error  Implicit any in catch clause                       @typescript-eslint/no-implicit-any-catch

✖ 16 problems (16 errors, 0 warnings)
```

Reply:
I had to go back to the add-linting commit anyway to address a commit cleanliness comment from Alex so I met in the middle here if that's cool with you: ran `lint:fix` over the original JS in the commit that adds eslint-config-xo and company, but does _not_ resolve the outstanding 18 (up from 16 because I'd missed `eslint-config-xo` not being inherently pulled in by `eslint-config-xo-typescript`, oops) non-auto-fixable errors. It does, however, resolve about 50 auto-fixable errors at that point in the commit history.

That's all now implemented in ecdacd5